lib/Backend/GlobOpt.cpp

```diff
@@ -16196,6 +16196,22 @@ GlobOpt::OptIsInvariant(
             allowNonPrimitives = true;
         }
         break;
+    case Js::OpCode::CheckObjType:
+        // Bug 11712101: If the operand is a field, ensure that its containing object type is invariant 
+        // before hoisting -- that is, don't hoist a CheckObjType over a DeleteFld on that object.
+        // (CheckObjType only checks the operand and its immediate parent, so we don't need to go
+        // any farther up the object graph.)
+        Assert(instr->GetSrc1());
+        PropertySym *propertySym = instr->GetSrc1()->AsPropertySymOpnd()->GetPropertySym();
+        Assert(propertySym != nullptr);  // QUESTION FOR REVIEWER: Can this ever be false?
```

Comment:
>Assert(propertySym != nullptr);  / [](start = 8, length = 34)

It cannot.  The As getter above could potentially fail, but I don't believe so.  You can remove this assert.